src/main/java/edu/harvard/iq/dataverse/DOIEZIdServiceBean.java

```diff
@@ -124,15 +124,16 @@ public boolean alreadyExists(DvObject dvObject) throws Exception {
      *
      * @param dvObject the Dataset whose metadata needs to be modified
      * @return the Dataset identifier, or null if the modification failed
+     * @throws java.lang.Exception
      */
-    
     @Override
     public String modifyIdentifierTargetURL(DvObject dvObject) throws Exception {
         String identifier = getIdentifier(dvObject);
         HashMap<String, String> metadata = new HashMap<>();
         metadata.put("_target", getTargetUrl(dvObject));
         try {
-            ezidService.setMetadata(identifier, metadata);
+            ezidService.setMetadata(identifier,
+                (metadata instanceof HashMap) ? (HashMap)metadata : new HashMap<>(metadata));
```

Comment:
Given line 132, don't we know that metadata is a HashMap? Given that, why bother with this check?

Reply:
Fixed.